### PR TITLE
Change `SstFileMetaData::size` from `size_t` to `uint64_t`

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: Because even 32-bit systems can have large files

This is a "change" that I don't want intermingled with an upcoming refactoring.

Test Plan: CI

### CODE REVIEW

**Review by zhichao-cao — APPROVED** [review]
LGTM

File: tools/db_bench_tool.cc [hunk 1]
```diff
@@ -5407,7 +5407,7 @@ class Benchmark {
         }
         if (levelMeta.level == 0) {
           for (auto& fileMeta : levelMeta.files) {
-            fprintf(stdout, "Level[%d]: %s(size: %" ROCKSDB_PRIszt " bytes)\n",
+            fprintf(stdout, "Level[%d]: %s(size: %" PRIi64 " bytes)\n",
```
Comment (mrambacher) [inline tools/db_bench_tool.cc:5410]: Shouldn't this be PRIu64?

Comment (pdillinger) [reply tools/db_bench_tool.cc:5410]: Yes, I was just copying from a few lines down. Doesn't matter in practice.

**Review by mrambacher — APPROVED** [review]
LGTM

**Pull request closed without merging**